core/internal/api/api.go

```diff
@@ -200,6 +200,9 @@ type ClientOptions struct {
 	//
 	// If Proxy is nil or returns a nil *URL, no proxy will be used.
 	Proxy func(*http.Request) (*url.URL, error)
+
+	// Function that get called before the retry attempt to prepare the request.
+	PrepareRetry func(*http.Request) error
```

Comment:
Is this used outside of tests? If not, it should be removed.

Reply:
The option itself is only used in tests, but I am a bit hesitant to remove since I think the test is useful. Open to other ideas, but maybe it could be done in a follow up PR